sound/soc/sof/intel/hda-dai-ops.c

```diff
@@ -237,15 +237,32 @@ static unsigned int generic_calc_stream_format(struct snd_sof_dev *sdev,
 					       struct snd_pcm_substream *substream,
 					       struct snd_pcm_hw_params *params)
 {
+	struct snd_soc_pcm_runtime *rtd = asoc_substream_to_rtd(substream);
+	struct snd_soc_dai *codec_dai;
 	unsigned int format_val;
+	u32 ch_mask = 0;
+	int num_channels;
+	int j;
 
-	format_val = snd_hdac_calc_stream_format(params_rate(params), params_channels(params),
+	/*
+	 * if the multiple dais are handled by the same dailink, we may need to update the
+	 * stream channel count - the params are modified in soc-pcm based on the codec_ch_maps info
+	 */
+	for_each_rtd_codec_dais(rtd, j, codec_dai)
+		ch_mask |= rtd->dai_link->codec_ch_maps[j].ch_mask;
+
+	num_channels = hweight_long(ch_mask);
+	if (num_channels != params_channels(params))
+		dev_dbg(sdev->dev, "configuring stream format for %d channels, params_channels was %d\n",
+			num_channels, params_channels(params));
```

Comment:
do we need this print?

Reply:
it's useful to visualize where the channel differences happen. I prefer to keep it, it helped me locate the configuration issue.